src/replicache.ts

```diff
@@ -244,6 +245,17 @@ export default class Replicache implements ReadTransaction {
     window.addEventListener('storage', this._onStorage);
   }
 
+  /**
+   * The client ID for this instance of Replicache. Each web browser and
+   * instance of Replicache gets a unique client ID keyed by the
+   * {@link ReplicacheOptions.name | name}. This is persisted locally between
+   * sessions (unless [[useMemstore]] is true in which case it is reset every
+   * time a new Replicache instance is created).
+   */
+  async clientID(): Promise<string> {
```

Comment:
Nit: What do you think about making this a read-only property? I feel like it better reflects the performance expectations.

Reply:
I had it like that before but I hate having to throw if the promise has not yet been resolved. Also, we do not expose the "opened" promise so it is hard to know when you can get the clientID. The following would throw:

```js
const rep = new Replicache();
console.log(rep.clientID);
```

One would have to do something cryptic like the following to ensure that replicache has been openened.

```js
const rep = new Replicache();
await rep.query(tx => null);
console.log(rep.clientID); 
```